### PR TITLE
Fix issue 36

### PR DESCRIPTION
**Pull request by devvv1337**
implemented the test of the deployment api( client/server) to test the modele accuracy ( must be >80 to validate the ci)

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (CirSandro) [inline .github/workflows/ci.yml:1]: No pass the CI

Don't change all the code in the file, just add your code where it's needed

File: src/__init__.py
Comment (CirSandro) [inline src/__init__.py:1]: Check with @Dixtawi if the file is not essential with the library (remains that on your missions, if you see something weird or that does not work = Issues)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,90 +9,199 @@ on:
...
+            fi
+            echo "Waiting for client... attempt $i"
+            sleep 1
+          done
+
+      # Run the tests
+      - name: Run tests
+        run: |
+          set -e
+          make test || (echo "Tests failed! Server log:" && cat server.log && echo "Client log:" && cat client.log && exit 1)
```
Comment (CirSandro) [inline .github/workflows/ci.yml:150]: Is it possible to display the accuracy on success?

**Comment by Dixtawi** [comment]
Seems like the CI is not working on your PR because you have a problem with dependencies. Check out the dependencies.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (devvv1337) [reply .github/workflows/ci.yml:127]: done

File: src/__init__.py
Comment (devvv1337) [reply src/__init__.py:1]: i put it back to be sure to not impact the hard work of @Dixtawi

File: .github/workflows/ci.yml
Comment (devvv1337) [reply .github/workflows/ci.yml:1]: i fixed the ci to use the correct python version for the run_notebook part

**Comment by Dixtawi** [comment]
matplotlib dependancy missing in requirements.txt it seems

**Comment by CirSandro** [comment]
> matplotlib dependancy missing in requirements.txt it seems

It's good normally the merge main fixes the problem I'll take care of it

**Review by Fahol0 — APPROVED** [review]

File: tests/test_api.py [hunk 2]
```diff
@@ -1,113 +1,174 @@
...
+        retry_delay=5,
+        success=False
+    )
+
+    for i in range(30):
+        if check_server():
+            break
+        if i == 29:
+            pytest.fail("Server not accessible after 30 attempts")
+        time.sleep(1)
```
Comment (Dixtawi) [inline tests/test_api.py:152]: isn't ths which is making the api slow?

**Review by Dixtawi — DISMISSED** [review]

File: tests/test_api.py (hunk 2 above)
Comment (Mkbrad77) [reply tests/test_api.py:155]: lt is a real possibility we have to fix that

**Review by Mkbrad77 — APPROVED** [review]

**Review by CirSandro — APPROVED** [review]

**Pull request merged**